invenio_users_resources/services/schemas.py

```diff
@@ -57,7 +57,8 @@ class UserSchema(BaseRecordSchema, FieldPermissionsMixin):
     # NOTE: API should only deliver users that are active & confirmed
     active = fields.Boolean()
     confirmed = fields.Boolean(dump_only=True)
-    is_current_user = fields.Method("is_self", dump_only=True)
+    # psaiz: I don't understand why I have to comment this out :(
+    # is_current_user = fields.Method("is_self", dump_only=True)
```

Comment:
I find this one confusing :(

Reply:
FYI, if I don't comment it out, I get 
```
Traceback (most recent call last):
  File "/Users/psaiz/github/invenio-administration/invenio_administration/marshmallow_utils.py", line 104, in jsonify_schema
    "type": custom_mapping[field_type_name],
KeyError: <class 'marshmallow.fields.Method'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/app.py", line 2552, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/werkzeug/middleware/proxy_fix.py", line 187, in __call__
    return self.app(environ, start_response)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/werkzeug/middleware/dispatcher.py", line 78, in __call__
    return app(environ, start_response)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/app.py", line 2532, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask_principal.py", line 199, in _decorated
    rv = f(*args, **kw)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
  File "/Users/psaiz/.local/share/virtualenvs/psaiz-master-PcfYpoPq/lib/python3.9/site-packages/flask/views.py", line 188, in dispatch_request
    return current_app.ensure_sync(meth)(**kwargs)
  File "/Users/psaiz/github/invenio-administration/invenio_administration/views/base.py", line 388, in get
    serialized_schema = self._schema_to_json(schema)
  File "/Users/psaiz/github/invenio-administration/invenio_administration/views/base.py", line 195, in _schema_to_json
    return jsonify_schema(schema)
  File "/Users/psaiz/github/invenio-administration/invenio_administration/marshmallow_utils.py", line 109, in jsonify_schema
    raise Exception(f"Unrecognised schema field {field}: {field_type_name}")
Exception: Unrecognised schema field is_current_user: <class 'marshmallow.fields.Method'>
```
when I go to /administration/users/

Reply:
I have a feeling that this field could be needed somewhere else (needs checking though), so probably it shouldn't be commented out.

The error seems to occur because `fields.Method` isn't listed in the `custom_mapping` defined in [invenio-administration](https://github.com/inveniosoftware/invenio-administration/blob/main/invenio_administration/marshmallow_utils.py#L22).

One approach could be to remove this field _definition_ altogether and just add the field value in the `@post_dump` method (given that `is_current_user` is `dump_only=True` anyway).
Then the value would still be available after dumping, and thus be available when other places need it.
On loading, it will be ignored anyway (b/c it's `dump_only`).
WDYT?

Reply:
Sorry, I'm kind of lost here :(. I've put the `@post_dump()` in front of the `is_self` function, and that didn't help. Then, it expects the is_self to receive one more parameter. Probably this is not what you meant.

I've been looking at the `communities`, since there is also a `is_current_user` [there](https://github.com/inveniosoftware/invenio-communities/blob/master/invenio_communities/members/services/schemas.py#L155). That one has one more level of indirection, since is_current_user points to the function is_current_user, which then points to the is_self.

I've also been looking at the places that currently use the `is_current_user`. The only ones I could find are in the tests. What am I missing here?

Reply:
@zzacharo  figured out that here what we were missing was something like:
```
psaiz@mac-psaiz invenio-administration % git diff invenio_administration/marshmallow_utils.py                     
diff --git a/invenio_administration/marshmallow_utils.py b/invenio_administration/marshmallow_utils.py
index e262be5..5ed3c4c 100644
--- a/invenio_administration/marshmallow_utils.py
+++ b/invenio_administration/marshmallow_utils.py
@@ -36,6 +36,7 @@ custom_mapping = {
     fields.Date: "date",
     fields.TimeDelta: "timedelta",
     fields.Decimal: "decimal",
+    fields.Method: "method",
```

Adding that indeed solves the issue. Now, since that's on a different repo, are we supposed to send first a PR to that repo, increase the version number, and then send this PR, including also the requirement on the new version of invenio_administration?

Reply:
we agreed with @ppanero to include the fields.Method in the mapping for the time being, but ignore it in the administration panel (not to implement any associated behaviour or display) since handling of the Method field is not needed now. We should open a ticket explaining this issue for the future (implementing the field.Method handler)

Reply:
Thanks for the update. What about doing something like:
```
+++ b/invenio_administration/marshmallow_utils.py
@@ -54,6 +54,8 @@ def jsonify_schema(schema):
 
         if is_links:
             continue
+        if isinstance(field_type, fields.Method):
+            continue
```
This might be cleaner that the mapping (which I totally understand why Pablo doesn't like it).

Reply:
I am in favor of the mapping solution just because is easy to detect any errors, it will give a clear exception in `jsonify_schema`! IF you create an issue then we tackle this in a later point in time ;)

Reply:
Ok, thanks @zzacharo. Let's see if https://github.com/inveniosoftware/invenio-administration/pull/175 is what you wanted. I've also created https://github.com/inveniosoftware/invenio-administration/issues/174 for the backlog